src/Bicep.Core/Semantics/ParamsSemanticModel.cs

```diff
@@ -118,39 +118,44 @@ public ImmutableArray<IDiagnostic> GetAllDiagnostics()
         /// </summary>
         public ParamFileSymbol Root => this.ParamBinder.ParamFileSymbol;
 
-        public static Uri? TryGetBicepFileUri(out ToListDiagnosticWriter allDiagnostics, IFileResolver fileResolver, BicepParamFile bicepParamFile)
+        public static Uri? TryGetBicepFileUri(out ImmutableArray<IDiagnostic> compilationLoadDiagnostics, IFileResolver fileResolver, BicepParamFile bicepParamFile)
         {
-            allDiagnostics = ToListDiagnosticWriter.Create();
+            var diagnosticsWriter = ToListDiagnosticWriter.Create();
             var usingDeclarations = bicepParamFile.ProgramSyntax.Children.OfType<UsingDeclarationSyntax>();
 
             if(usingDeclarations.FirstOrDefault() is not {} usingDeclaration)
             {
-                allDiagnostics.Write(new TextSpan(0, 0), x => x.UsingDeclarationNotSpecified());
+                diagnosticsWriter.Write(new TextSpan(0, 0), x => x.UsingDeclarationNotSpecified());
+                compilationLoadDiagnostics = diagnosticsWriter.GetDiagnostics().ToImmutableArray();
                 return null;
             }
             
             if(usingDeclarations.Count() > 1)
             {
                 foreach(var declaration in usingDeclarations)
                 {
-                    allDiagnostics.Write(declaration.Keyword, x => x.MoreThanOneUsingDeclarationSpecified());
+                    diagnosticsWriter.Write(declaration.Keyword, x => x.MoreThanOneUsingDeclarationSpecified());
```

Comment:
This part is a bit repetitive in this function but I think that should be fine?

Reply:
I agree, this level of repetition is fine.